tests/slr_integration/Service/Seat_Types_Test.php

```diff
@@ -269,4 +269,477 @@ public function test_update_tickets_capacity(): void {
 		$this->assertEquals( 266, get_post_meta( $post_3_ticket_3, $capacity_meta_key, true ) );
 		$this->assertEquals( 266, get_post_meta( $post_3_ticket_3, '_stock', true ) );
 	}
-}
\ No newline at end of file
+
+	public function test_update_tickets_with_calculated_stock_and_capacity_simple(): void {
+		/** @var \Tribe__Tickets__Tickets_Handler $tickets_handler */
+		$tickets_handler   = tribe( 'tickets.handler' );
+		$capacity_meta_key = $tickets_handler->key_capacity;
+		// Create the 5 seat types.
+		Seat_Types::insert_rows_from_service(
+			[
+				[
+					'id'       => 'seat-type-uuid-1',
+					'name'     => 'Seat Type 1',
+					'seats'    => 10,
+					'mapId'    => 'map-uuid-1',
+					'layoutId' => 'layout-uuid-1',
+				],
+				[
+					'id'       => 'seat-type-uuid-2',
+					'name'     => 'Seat Type 2',
+					'seats'    => 20,
+					'mapId'    => 'map-uuid-1',
+					'layoutId' => 'layout-uuid-1',
+				]
+			]
+		);
+
+		$seat_types = tribe( Seat_Types::class );
+
+		[
+			$post_id_1,
+			$post_id_2,
+			$post_id_3,
+			$post_id_4,
+		] = static::factory()->post->create_many( 4 );
+
+		// Post 1 set up.
+		$post_1_ticket_1 = $this->create_tc_ticket( $post_id_1, 10 );
+		$post_1_ticket_2 = $this->create_tc_ticket( $post_id_1, 20 );
+
+		update_post_meta( $post_id_1, Meta::META_KEY_LAYOUT_ID, 'layout-uuid-1' );
+		update_post_meta( $post_id_1, $capacity_meta_key, 30 );
+		update_post_meta( $post_id_1, Global_Stock::GLOBAL_STOCK_LEVEL, 30 );
+
+		update_post_meta( $post_1_ticket_1, Meta::META_KEY_SEAT_TYPE, 'seat-type-uuid-1' );
+		update_post_meta( $post_1_ticket_1, $capacity_meta_key, 10 );
+		update_post_meta( $post_1_ticket_1, '_stock', 10 );
```

Comment:
```suggestion
		update_post_meta( $post_1_ticket_1, '_stock', 10 );
```

Instead of updating the `_stock` like this we can also use args inside `create_ticket` like [this](https://github.com/the-events-calendar/event-tickets/blob/66a92ccc37ab2d0b1f450e6b1de001f5b20bace2/tests/commerce_integration/TEC/Tickets/Commerce/Stock/SharedCapacity/SharedCapacityTest.php#L60)

It uses the original API that is used for adding ticket and should emulate the real scenario better.

Reply:
I did try out and even though i think its good practice. At the end, i believe it makes the tests more difficult to read (we have to play around with the variable passed into `create_tc_ticket` since the capacity is changing) and we would have to continue manually updating `_stock` in order to fake ticket with sales.

Reply:
For making sales we usually do make use of the `create_order` that way it emulates the sales. I understand you are taking a more unit test approach for this one which should be fine.